app/src/lib/stores/git-store.ts

```diff
@@ -1497,6 +1499,21 @@ export class GitStore extends BaseStore {
     })
   }
 
+  public async discardChangesFromSelection(
+    filePath: string,
+    diff: ITextDiff,
+    selection: DiffSelection
+  ) {
+    await this.performFailableOperation(async () => {
+      await discardChangesFromSelection(
```

Comment:
I think we can skip the inner await here since `performFailableOperation` will await it so it only creates another Promise closure 

```suggestion
      discardChangesFromSelection(
```

Reply:
I don't think we can just remove the `await` here: we'll need to also return the promise that gets returned by `discardChangesFromSelection()` to make `performFailableOperation()` await that promise.

I can do that by removing the braces from the function and get advantage of the implicit return.